airflow/models/baseoperator.py

```diff
@@ -1081,13 +1080,16 @@ def get_extra_links(self, dttm, link_name):
         else:
             return None
 
-
-# pylint: disable=protected-access
-BaseOperator._serialized_fields = frozenset(
-    vars(BaseOperator(task_id='test')).keys() - {
-        'inlets', 'outlets', '_upstream_task_ids', 'default_args'
-    } | {'_task_type', 'subdag', 'ui_color', 'ui_fgcolor', 'template_fields'}
-)
+    @classmethod
+    def get_serialized_fields(cls):
+        """Stringified DAGs and operators contain exactly these fields."""
+        if not cls._serialized_fields:
+            cls._serialized_fields = frozenset(
+                vars(BaseOperator(task_id='test')).keys() - {
```

Comment:
I could have sworn I tried this (putting this in the class and couldn't get it working.) Much nicer!

(Shame that cached_property doesn't work on classes.)

Reply:
I yet have to run all tests. but I hope it will work!